### PR TITLE
Implement minimum viable product

### PR DESCRIPTION
**Pull request by ivan-gomes**

### CODE REVIEW

File: lib/view-service-stack.js [hunk 1]
```diff
@@ -1,17 +1,361 @@
...
-   * @param {cdk.Construct} scope
-   * @param {string} id
-   * @param {cdk.StackProps=} props
-   */
-  constructor(scope, id, props) {
-    super(scope, id, props);
-
-    // The code that defines your stack goes here
-  }
+const {Construct, Duration, RemovalPolicy, Stack, StackProps} = require("@aws-cdk/core");
```
Comment (blake-regalia) [inline lib/view-service-stack.js]: this is nitpicking, but it would be nice to use single-quote strings here to be consistent with rest of script

**Review by HuiJun — COMMENTED** [review]
This is also a nitpick. It seems like this is trying to create a lot of roles, lambdas, groups etc. I think it would be useful to namespace any persistent ones that do get created.

**Review by HuiJun — DISMISSED** [review]

**Comment by ivan-gomes** [comment]
> This is also a nitpick. It seems like this is trying to create a lot of roles, lambdas, groups etc. I think it would be useful to namespace any persistent ones that do get created.

CloudFormation generates physical IDs like `ViewServiceStack-ClearNeptuneGraph########-############` and assigns tags like `aws:cloudformation:stack-name=ViewServiceStack`. Let's make sure we're good with this during the technical deep dive.

File: lib/view-service-stack.js (hunk 1 above)
Comment (ivan-gomes) [reply lib/view-service-stack.js]: Thanks, fixed!

**Review by HuiJun — APPROVED** [review]

File: src/functions.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    });
+
+    const writeStream = fs.createWriteStream(fileObj.name);
+    const exportConfig = _buildExportConfig({
+        ...event?.page,
+        output: writeStream
+    });
+
+    await wikiExport(exportConfig);
+    writeStream.close();
```
Comment (blake-regalia) [inline src/functions.js]: the `close` method should not be called from user-code on readable streams

File: src/functions.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    };
+}
+
+function _hashFile(algorithm, stream) {
+    return new Promise((resolve, reject) => stream
+        .on('error', reject)
+        .pipe(crypto.createHash(algorithm)
+            .setEncoding('hex'))
+        .once('finish', function () {
+            resolve(this.read())
```
Comment (blake-regalia) [inline src/functions.js]: this function is a bit hacky and not guaranteed to work as intended

File: src/functions.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+function _buildImportConfig(options) {
+    return {
+        ...options.graph && {graph: options.graph},
+        ...options.prefix && {prefix: options.prefix}
+    }
+}
+
+// TODO Revert after upstream fix
+async function _wikiChildPages(options) {
+    return wikiChildPages(options).then(children => children.map(child => !options?.as_urls ? parseInt(child) : child));
```
Comment (blake-regalia) [inline src/functions.js]: what is the purpose of `parseInt` here? why does the page id need to be a javascript number?

File: src/functions.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    return {
+        stack: stack,
+        limit: limit,
+        result: result,
+        ...event.server && {server: event.server},
+        ...event.as_urls && {as_urls: event.as_urls}
+    }
+}
+
+exports.generateUuid = async function (_event) {
```
Comment (blake-regalia) [inline src/functions.js]: these functions don't need to be async. best practice is to only use async if `await` keyword is used.

**Review by blake-regalia — CHANGES_REQUESTED** [review]

File: src/functions.js (hunk 3 above)
Comment (blake-regalia) [reply src/functions.js]: i can take care of this one

**Review by blake-regalia — APPROVED** [review]

**Comment by ivan-gomes** [comment]
LGTM

**Pull request merged**